mantidproject/mantid
# Poisson Cost Function

**Description of work.**
This finishes Roman's work on Poisson cost functions (started in #25686)

@alicerussell1 could you add release notes if required, as you are more familiar with the potential benefits of this technique.

Changes since the original PR:
- Squashes the original commit series and resolve conflicts with master
- Added unit tests for the methods
- Moved protected methods into public following the conventions of the base classes
- Fixed a bug where using any parameters would result in an OOB
- Fixed a bug where the determinant was always calculated instead of respecting the flag

**To test:**
- Ensure unit tests pass

```
Load some muon data and plot a few spectra
Fit to the spectra using least squares and poisson cost functions
The parameters should be similar but different.
```



Fixes #14490 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
